lib/miq_expression/relative_datetime.rb

```diff
@@ -32,23 +32,33 @@ def self.normalize(rel_time, tz, mode = "beginning")
 
       if interval == "quarters"
         ts = Time.now.in_time_zone(tz).beginning_of_quarter
-        (ts - (value.to_i * 3.months)).send("#{mode}_of_quarter")
+        to_date((ts - (value.to_i * 3.months)).send("#{mode}_of_quarter"), is_date)
       else
-        value.to_i.send(interval).ago.in_time_zone(tz).send("#{mode}_of_#{interval.singularize}")
+        to_date(value.to_i.send(interval).ago.in_time_zone(tz).send("#{mode}_of_#{interval.singularize}"), is_date)
       end
     elsif rt == "today"
-      Time.now.in_time_zone(tz).send("#{mode}_of_day")
+      to_date(Time.now.in_time_zone(tz).send("#{mode}_of_day"), is_date)
     elsif rt == "yesterday"
-      1.day.ago.in_time_zone(tz).send("#{mode}_of_day")
+      to_date(1.day.ago.in_time_zone(tz).send("#{mode}_of_day"), is_date)
     elsif rt == "now"
       t = Time.now.in_time_zone(tz)
-      mode == "beginning" ? t.beginning_of_hour : t.end_of_hour
+      to_date(mode == "beginning" ? t.beginning_of_hour : t.end_of_hour, is_date)
     else
       # Assume it's an absolute date or time
       value_is_date = !rel_time.include?(":")
       ts = Time.use_zone(tz) { Time.zone.parse(rel_time) }
       ts = ts.send("#{mode}_of_day") if mode && value_is_date
-      ts
+      to_date(ts, is_date)
+    end
+  end
+
+  def self.to_date(value, is_date)
```

Comment:
I'm wondering if `to_date` is confusing because we're dealing with dates vs times. Perhaps `coerce`?